demos/vc_issuer/app/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { AuthClient } from "@dfinity/auth-client";
+
+import React, { useState } from "react";
+import ReactDOM from "react-dom/client";
+
+import { VcIssuer } from "./issuer";
+
+import "./main.css";
+
+/** Reads the canister ID from the <script> tag.
+ *
+ * The canister injects the canister ID as a `data-canister-id` attribute on the script tag, which we then read to figure out where to make the IC calls.
+ */
+const readCanisterId = (): string => {
+  // The backend uses a known element ID so that we can pick up the value from here
+  const setupJs = document.querySelector(
+    "[data-canister-id]"
+  ) as HTMLElement | null;
+  if (!setupJs || setupJs.dataset.canisterId === undefined) {
+    throw new Error("canisterId is undefined"); // abort further execution of this script
+  }
+
+  return setupJs.dataset.canisterId;
+};
+
+/** The issuer management app */
+const App = () => {
+  // A global "disabled" flag that gets set during async operations
+  // to prevent the user to interact with the app until the operation
+  // is completed.
+  const [dsbld, setDsbld] = useState<boolean>(false);
+
+  // Run an async op while disabling buttons/interaction
+  const withDisabled = async <A, _>(fn: () => Promise<A>): Promise<A> => {
+    setDsbld(true);
+    try {
+      return await fn();
+    } finally {
+      setDsbld(false);
+    }
+  };
+
+  // The principal, set during auth
+  const [principal, setPrincipal] = useState<string | undefined>(undefined);
+
+  // The canister logs
+  const [canisterLogs, setCanisterLogs] = useState<string[]>([]);
+
+  // The URL used for connecting to the IDP
+  const [iiUrl, setIiUrl] = useState<string>("http://localhost:5173");
+
+  // Perform authentication with IDP and show principal
+  const authAndShow = () =>
+    withDisabled(async () => {
+      const authClient = await AuthClient.create();
+
+      await new Promise<void>((resolve, reject) => {
+        authClient.login({
+          identityProvider: iiUrl,
+          onSuccess: () => resolve(),
+          onError: reject,
+        });
+      });
+
+      setPrincipal(authClient.getIdentity().getPrincipal().toText());
+    });
+
+  // Add the principal as employee
+  const addEmployee = (principal: string) =>
+    withDisabled(async () => {
+      const canisterId = readCanisterId();
+      const vcIssuer = new VcIssuer(canisterId);
+      const res: string = await vcIssuer.addEmployee({ principal });
+      setCanisterLogs([...canisterLogs, res]);
+    });
+
+  return (
+    <main data-page="add-employee">
+      <h1>Add Employee</h1>
+      <section>
+        <label>
+          Principal:
+          {principal ? (
+            <output data-role="principal">{principal}</output>
+          ) : (
+            <output data-role="principal" data-unset>
+              not authenticated
+            </output>
+          )}
+        </label>
+      </section>
+      <section>
+        <label>
+          Canister Logs:
```

Comment:
Nit: Maybe rename to canister interaction log? Otherwise, this might suggest that you are somehow getting to the log output of the canister itself.

Reply:
Makes sense! I've renamed it to "Canister Replies" although I'm not sure it's much clearer; I find "Canister Interaction Logs" a bit long. I'll see if I get a better idea over the next few days.